comp/trace/agent/run.go

```diff
@@ -41,7 +42,7 @@ import (
 )
 
 // runAgentSidekicks is the entrypoint for running non-components that run along the agent.
-func runAgentSidekicks(ctx context.Context, cfg config.Component, telemetryCollector telemetry.TelemetryCollector) error {
+func runAgentSidekicks(ctx context.Context, cfg config.Component, statsd statsd.Component, telemetryCollector telemetry.TelemetryCollector) error {
```

Comment:
Can we move the statsd initialization, as it's just one line, outside of this function?  When I created it I assumed it was temporary code and that other components injected in the tracer would be already initialized.

As far as I can tell, we could initialize this anywhere else where config.Component is available, as the tags arguments only depends in a global variable `version.AgentVersion`. It could be similar to what I did with the `telemetry.TelemetryCollector` value:

```go
fx.Provide(func(cfg config.Component) telemetry.TelemetryCollector { return telemetry.NewCollector(cfg.Object()) })
```